File: frontend/src/app/models/projects/project-response-dto.ts
import { TagDto } from '../tags/tag-dto';

import { DbEngine } from './db-engine';

export interface ProjectResponseDto {
    id: number;
    isAuthor: boolean;
    name: string;
    description: string | null;
    dbEngine: DbEngine;
    createdAt: Date;

    tags: TagDto[];
}

File: backend/Squirrel.Core/Squirrel.Core.Common/DTO/Project/ProjectResponseDto.cs
using Squirrel.Core.Common.DTO.Tag;
using Squirrel.Core.DAL.Enums;

namespace Squirrel.Core.Common.DTO.Project;

public sealed class ProjectResponseDto
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public DbEngine DbEngine { get; set; }
    public DateTime CreatedAt { get; set; }

    public ICollection<TagDto>? Tags { get; set; }
}